code/game/objects/structures/mirror.dm

```diff
@@ -8,6 +10,30 @@
 	anchored = 1
 	var/shattered = 0
 
+	/// Visual object for handling the viscontents
+	var/datum/weakref/ref
+	vis_flags = VIS_HIDE
+	var/timerid = null
+
+/obj/structure/mirror/New()
+	var/obj/effect/reflection/reflection = new(src.loc)
+	reflection.setup_visuals(src)
+
+	ref = makeweakref(reflection)
```

Comment:
Does it make sense to use a weakref here? Do you expect the reflection to be deleted?